### PR TITLE
Prep work for running bazel in prober:

### PR DESCRIPTION
**Pull request by vadimberezniker**
- Add base container image that has the deps that Bazel needs to run
 - Add Bazel 4.1 binaries (and also bumped Bazel 3.7 to latest minor version)
 - Split server/testutil/bazel into test (server/testutil/testbazel) and
   non-test (server/util/bazel) packages.

---

**Version bump**: None

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -2659,15 +2659,27 @@ def install_buildbuddy_dependencies(workspace_name = "buildbuddy"):
...
     )
     http_file(
-        name = "io_bazel_bazel-3.7.0-linux-x86_64",
-        sha256 = "b7583eec83cc38302997098a40b8c870c37e0ab971a83cb3364c754a178b74ec",
-        urls = ["https://github.com/bazelbuild/bazel/releases/download/3.7.0/bazel-3.7.0-linux-x86_64"],
+        name = "io_bazel_bazel-3.7-linux-x86_64",
+        sha256 = "70dc0bee198a4c3d332925a32d464d9036a831977501f66d4996854ad4e4fc0d",
+        urls = ["https://github.com/bazelbuild/bazel/releases/download/3.7.2/bazel-3.7.2-linux-x86_64"],
+        executable = True,
+    )
```
Comment (bduffany) [inline deps.bzl:2672]: Do the tests pass if globally replacing 3.7 with 4.1? (I don't think there's a specific reason I chose 3.7 originally, other than that it matched our .bazelversion, so I'd be fine to use 4.1 everywhere)

File: enterprise/server/test/integration/ci_runner/ci_runner_test.go [hunk 2]
```diff
@@ -231,7 +231,7 @@ func TestCIRunner_WorkspaceWithCustomConfig_RunsAndUploadsResultsToBES(t *testin
 	runnerInvocation := res.Invocation[0]
 	// Since our workflow just runs `bazel version`, we should be able to see its
 	// output in the action logs.
-	assert.Contains(t, runnerInvocation.ConsoleBuffer, "Build label: 3.7.0")
+	assert.Contains(t, runnerInvocation.ConsoleBuffer, "Build label: 3.7")
```
Comment (bduffany) [inline enterprise/server/test/integration/ci_runner/ci_runner_test.go:234]: Maybe remove the "3.7" entirely so we don't have to maintain this in the future (the test is only trying to assert that the `bazel version` command executes, not necessarily that it uses any particular version)

Comment (vadimberezniker) [reply enterprise/server/test/integration/ci_runner/ci_runner_test.go]: Done

File: deps.bzl (hunk 1 above)
Comment (vadimberezniker) [reply deps.bzl:2672]: I want to test with whatever versions customers are using out in the wild (which could be 3.7 & 4.1), but I don't have any good sense of what customers are actually using today. Maybe @tylerwilliams  knows.
Unfortunately it doesn't look like this information is easy to query right now from our DB.

**Review by tylerwilliams — APPROVED** [review]
LGTM

File: deps.bzl (hunk 1 above)
Comment (vadimberezniker) [reply deps.bzl:2672]: Done, https://github.com/buildbuddy-io/buildbuddy-internal/issues/630

**Pull request merged**